src/router.js

```diff
@@ -13,9 +13,14 @@ import {
   isFalse
 } from './utils';
 
+let IEoldURL = '';
+
 function handleRouteChange(e) {
-  const oldPath = pathFromHash(e.oldURL);
-  const newPath = pathFromHash(e.newURL);
+  const oldURL = e.oldURL || IEoldURL;
+  const newURL = e.newURL || window.location.hash;
```

Comment:
this was my original approach. the problem with this is that the hash could be changed multiple times before the event is actually handled, causing the new url to be wrong and causing handlers to fire incorrectly.